src/communication/v2/server.hpp

```diff
@@ -75,11 +77,11 @@ class Server final {
          size_t workers_count = std::thread::hardware_concurrency())
       : endpoint_{endpoint},
         service_name_{service_name},
-        context_pool_{workers_count},
-        listener_{Listener<TSession, TSessionData>::Create(context_pool_, session_data, server_context, endpoint_,
-                                                           service_name_, inactivity_timeout_sec)} {}
+        context_thread_pool_{workers_count},
+        listener_{Listener<TSession, TSessionData>::Create(context_thread_pool_, session_data, server_context,
+                                                           endpoint_, service_name_, inactivity_timeout_sec)} {}
 
-  ~Server() { MG_ASSERT(!context_pool_.IsRunning(), "Server wasn't shutdown properly"); }
+  ~Server() { MG_ASSERT(!IsRunning(), "Server wasn't shutdown properly"); }
```

Comment:
Here I still question the decision, but I am more okay with this, as this server is quite specific compared to `IOContextThreadPool`. Also this worked in the same way as before and because we want to maintain "plug-and-play" compatibility, this is kind of necessary.